api/sync.js

```
@@ -62,8 +82,110 @@ function requireAuth(req, res, next) {
+    entry.windowStart = now;
+  }
+  entry.count++;
+  _rateLimits.set(userId, entry);
+  if (entry.count > RATE_LIMIT_MAX) {
+    return res.status(429).json({ error: 'Too many requests. Please wait before syncing again.' });
+  }
+  next();
+}
+
+// ─── Input Validation ─────────────────────────────────────────────────────────
+const ID_RE = /^[a-zA-Z0-9_\-]{1,128}$/;
```

ESLint

[error] Unnecessary escape character: \-.

(no-useless-escape)